annotations/views/rest_views.py

```diff
@@ -493,8 +500,76 @@ def _relabel(datum):
                 'identifier': 'uri'
             }
 
-            return {_fields.get(k, k): v for k, v in datum.iteritems()}
-        return Response({'results': map(_relabel, [c.data for c in concepts])})
+            return {_fields.get(k, k): v for k, v in datum.iteritems() }
+        results = map(_relabel, [c.data for c in concepts])
+        
+        for result in results:
+            if result["identities"]: # if identities exist append the first identitiy to the list so that we can filter out other identities against it
+                identities = result["identities"][0]["concepts"]
+            else:
+                identities = []
+            for ident in result["identities"]: # go through the identities in each result
+                for identity in identities: # go through the identities in the identities list
+                    if identity not in ident["concepts"]: # if the ideneities list does not contain the identity from the result then add it to the list
+                        identities.extend(ident["concepts"])
```

Comment:
I'm a little confused about what line 507-514 are supposed to do. What is the goal? That you have a list with all possible URIs for that result? E.g. ["http://viaf.org/viaf/85363862", "http://www.digitalhps.org/concepts/CONpeSHC70qxNC0"] for jean bacon?
 Couldn't you just iterate over all identity objects and then all concepts and add the URIs to a set?

Reply:
also, calling your list of concepts in line 508 'identities' is really confusing.